### DISCUSSION TITLE
Could not connect to this Shlink server" when adding a server at https://app.shlink.io

### DISCUSSION THREAD

**Discussion by jimtyrro** [shlinkio/discussions#14] [Help wanted] [answered]
Getting the following error when trying to connect to https://app.shlink.io.

CLI is working partially, can create short URLs but they are invalid. Getting the **502 Bad Gateway error.**

The developer's tools panel shows this error:

`GET https://s.tovp.org/rest/v3/health net::ERR_FAILED

fetchCallImpl @ main.js?attr=l7hz6hw…eVDnMj679Lbuyw:4956
fetch @ main.js?attr=l7hz6hw…eVDnMj679Lbuyw:4962
jsonRequest @ index-5iXwab9J.js:851
performRequest @ index-5iXwab9J.js:79
health @ index-5iXwab9J.js:79
(anonymous)@ index-5iXwab9J.js:945
(anonymous)@ index-5iXwab9J.js:945
(anonymous)@ index-5iXwab9J.js:851
(anonymous)@ index-5iXwab9J.js:945
(anonymous)@ index-5iXwab9J.js:851
(anonymous)@ index-5iXwab9J.js:851
(anonymous)@ index-5iXwab9J.js:851
Dq.n.<computed> @ index-5iXwab9J.js:48
(anonymous) @ index-5iXwab9J.js:945
D0 @ index-5iXwab9J.js:40
Gu @ index-5iXwab9J.js:40
(anonymous) @ index-5iXwab9J.js:40
E @ index-5iXwab9J.js:25
F @ index-5iXwab9J.js:25`

There's no logs to check to further investigate the issue. 

Any ideas how to troubleshoot this ?

Thank you.

**Comment by acelaya** [comment, marked as answer]
That's an incorrect nginx reverse proxy configuration. You should find more in nginx logs.

**Comment by jimtyrro** [comment]
Thank you Alex for the tip, I fixed this error, everything is working fine now. 

Question if I want to switch from Nginx runtime to Road Runner, do I have to remove or edit any of Nginx configs?

**Reply by acelaya** [reply]
RoadRunner is an app server, and provides its own HTTP server. You can still use nginx as reverse proxy, in case you want to configure things like HTTPS and such.

**Comment by jimtyrro** [comment]
And what's the best way to protect public access to the web-client if I'm using my own? Otherwise anyone can create any links without my authorization.

**Reply by acelaya** [reply]
> Otherwise anyone can create any links without my authorization.

That's actually not true. They need the API key.

**Reply by jimtyrro** [reply]
I think I understood. Thank you.

**Reply by acelaya** [reply]
You can find more information, context and future plans here https://github.com/shlinkio/shlink-web-client/issues/338
